README.md

```diff
@@ -64,141 +64,129 @@ RPC as well.
-
-The capability to discover event schemata decouples the provider of the schema
-from the service using it (e.g. environment, configuration files, or other
-services).
-
-Events get augmented with additional information when emitted. Every event gets
-an ID that is used for correlation in tracing by passing it on in events that
-are related down the line. If an event is the start of the chain then a new ID
-is generated. Additional metadata includes the timestamp and version).
-
-Events can also be emitted as dicts. Those will be automatically serialized but
-will get dropped by consumers that fail to validate them.
-
-    import twyla.service.events as events
-    import twyla.service.message as message
-    import twyla.service.test.helpers as helpers
-
-    payload = {
-        'message_type': 'integration',
-        'tenant': 'test-tenant',
-        'bot_slug': 'test-slug',
-        'channel': 'test-channel',
-        'channel_user_id': 'test-user-id',
-        'content': {}
+- `content: dict`: The event contents which will be serialized to JSON and
+  transmitted to listeners.
+
+- `context: dict`: Contextual data that is necessary for the proper processing
+  of the content.
+
+- `meta: dict`: Meta-information for correlation and debugging.
+
+The `meta` field is generated automatically, and contains the following:
+
+- `version: int`: The version of the message protocol
+
+- `timestamp`: The time the event was generated
+
+- `session_id`: A random UUID for tracing
+
+
+### Validating Events
+
+Both listeners and producers have to set validators for the events processed by
+twyla.service. For this purpose, `twyla.service.message.get_schemata` can be
+used to set schemata in the [JSONSchema](json-schema.org) format, as in the
+following sample:
+
+```Python
+from twyla.service import event
+
+content_schema = {
+    "$schema": "http://json-schema.org/draft-06/schema#",
+    "title": "Content",
+    "description": "User input",
+    "type": "object",
+    "properties": {
+        "emission": {"description": "Some emission",
+                     "type": "string"},
+        "user_id": {"description": "ID of the user",
+                    "type": "string"}
     }
+}
+
+content_schema_set = {
+    'api.user_input': content_schema
+}
+
+
+context_schema = {
+    "$schema": "http://json-schema.org/draft-06/schema#",
+    "title": "Context",
+    "description": "Some context",
+    "type": "object",
+    "properties": {
+        "tenant": {"description": "The tenant",
+                   "type": "string"},
+        "channel-id": {"description": "ID of communication channel",
+                       "type": "integer"}
+    }
+}
 
-    helpers.aio_run(events.emit('test-event', payload))
+event.set_schemata(content_schema_set, context_schema)
+```
 
+twyla.service will then validate both incoming and outgoing events from a
+service during operation.
 
-### Listening to Events
+### Raising Events
```

Comment:
Small nitpick regarding the README table of contents: the ToC link should be changed from `[Creating Events](#creating-events)` to `[Raising Events](#raising-events)` to prevent dead links.